libr/bin/format/xtac/bin_xtac.c

```diff
@@ -0,0 +1,587 @@
+	const ut32 p_addr_pair = bin->header->ptr_to_addr_pairs;
+	if (!(bin->address_pairs = R_NEWS0 (X86ArmAddrPair, n_addr_pairs))) {
+		r_sys_perror ("R_NEWS0 (xtac address pair)");
+		return false;
+	}
+	if (r_buf_read_at (bin->b, p_addr_pair, (ut8*) bin->address_pairs, addr_pair_size) < 0) {
+		R_LOG_WARN ("Warning: read (xtac address pairs)\n");
+		free (bin->address_pairs);
+		bin->address_pairs = NULL;
+		return false;
+	}
+	return true;
+}
+
+static bool r_bin_xtac_read_module_name(RBinXtacObj *bin) {
+	const ut32 len_of_mod_name = bin->header->size_of_mod_name / sizeof (ut16) + 1;
+	const ut32 p_mod_name = bin->header->ptr_to_mod_name;
+	if (!(bin->mod_name_u16 = R_NEWS0 (ut16, len_of_mod_name))) {
+		r_sys_perror ("R_NEWS0 (xtac module name)");
+		return false;
+	}
+	if (r_buf_read_at (bin->b, p_mod_name, (ut8*) bin->mod_name_u16, bin->header->size_of_mod_name) < 0) {
+		R_LOG_WARN ("Warning: read (xtac module name)\n");
+		free (bin->mod_name_u16);
+		bin->mod_name_u16 = NULL;
+		return false;
+	}
+	if (!(bin->mod_name_u8 = r_str_utf16_decode((ut8*) bin->mod_name_u16, bin->header->size_of_mod_name))) {
+		free (bin->mod_name_u16);
+		bin->mod_name_u16 = NULL;
+		return false;
+	}
+	return true;
+}
+
+static bool r_bin_xtac_read_nt_native_pathname(RBinXtacObj *bin) {
+	const ut32 len_of_nt_pname = bin->header->size_of_nt_pname / sizeof (ut16) + 1;
+	const ut32 p_nt_name = bin->header->ptr_to_nt_pname;
+	if (!(bin->nt_path_name_u16 = R_NEWS0 (ut16, len_of_nt_pname))) {
+		r_sys_perror ("R_NEWS0 (xtac nt path)");
+		return false;
+	}
+	if (r_buf_read_at (bin->b, p_nt_name, (ut8*) bin->nt_path_name_u16, bin->header->size_of_nt_pname) < 0) {
+		R_LOG_WARN ("Warning: read (xtac nt path)\n");
+		free (bin->nt_path_name_u16);
+		bin->nt_path_name_u16 = NULL;
+		return false;
+	}
+
+	char *nt_path_name_u8_raw = NULL;
+	if (!(nt_path_name_u8_raw = r_str_utf16_decode((ut8*) bin->nt_path_name_u16, bin->header->size_of_nt_pname))) {
+		free (bin->nt_path_name_u16);
+		bin->nt_path_name_u16 = NULL;
+		return false;
+	}
+
+	if (!(bin->nt_path_name_u8 = r_str_escape_utf8_for_json (nt_path_name_u8_raw, -1))) {
+		free (bin->nt_path_name_u16);
+		bin->nt_path_name_u16 = NULL;
+		free (nt_path_name_u8_raw);
+		bin->nt_path_name_u8 = NULL;
+		return false;
+	}
+
+	free (nt_path_name_u8_raw);
+
+	return true;
+}
+
+static bool r_bin_xtac_read_blck_stubs(RBinXtacObj *bin) {
+	if (!(bin->blck_stubs = r_list_newf(free))) {
+		r_sys_perror ("r_list_newf (r_bin_xtac_read_blck_stubs)");
+		return false;
+	}
+	const ut32 max_depth = 20;
+	ut32 i = 0;
+	ut32 p_blck_stub = bin->header->ptr_to_head_blck_stub;
+	do {
+		RBinBlckStubHeader *blck_stub = NULL;
+		if (!(blck_stub = R_NEW0 (RBinBlckStubHeader))) {
+			r_sys_perror ("R_NEW0 (xtac BLCK stub)");
+			return false;
+		}
+		if (r_buf_read_at (bin->b, p_blck_stub, (ut8*) blck_stub, sizeof (RBinBlckStubHeader) - sizeof (ut32)) < 0) {
+			R_LOG_WARN ("Warning: read (xtac BLCK stub)\n");
```

Comment:
```suggestion
			R_LOG_WARN ("read (xtac BLCK stub)");
```

R_LOG macros add the Warning and the newline already